app-libs/exchange-oracle/src/coingecko.rs

```diff
@@ -92,24 +111,49 @@ impl RestPath<String> for CoinGeckoMarket {
 	}
 }
 
-impl GetExchangeRate for CoinGeckoClient {
+impl<OCallApi> GetExchangeRate for CoinGeckoClient<OCallApi>
+where
+	OCallApi: EnclaveMetricsOCallApi,
+{
 	fn get_exchange_rate(&mut self, trading_pair: TradingPair) -> Result<ExchangeRate, Error> {
+		self.update_metric(
+			ExchangeRateOracleMetric::NumberRequestsIncrement(Self::metric_source()),
+		);
+
 		let fiat_id = self.fiat_currency_id(trading_pair.clone())?;
 		let crypto_id = self.crypto_currency_id(trading_pair.clone())?;
+
+		let timer_start = Instant::now();
+
 		let response = self
 			.client
 			.get_with::<String, CoinGeckoMarket>(
 				COINGECKO_PATH.to_string(),
 				&[(COINGECKO_PARAM_CURRENCY, &fiat_id), (COINGECKO_PARAM_COIN, &crypto_id)],
 			)
 			.map_err(Error::RestClient)?;
+
+		self.update_metric(ExchangeRateOracleMetric::ResponseTime(
+			Self::metric_source(),
+			timer_start.elapsed().as_millis(),
+		));
+
 		let list = response.0;
 		if list.is_empty() {
 			error!("Got no market data from coinGecko. Check params {:?} ", trading_pair);
 			return Err(Error::NoValidData)
 		}
+
 		match list[0].current_price {
-			Some(r) => Ok(ExchangeRate::from_num(r)),
+			Some(r) => {
+				let exchange_rate = ExchangeRate::from_num(r);
+				self.update_metric(ExchangeRateOracleMetric::ExchangeRate(
+					Self::metric_source(),
+					trading_pair.key(),
+					exchange_rate,
+				));
+				Ok(exchange_rate)
```

Comment:
Store the exchange rate itself in a metric too (grouped by source and trading pair)